### ISSUE TITLE
Better support for QuickTime video

### ISSUE THREAD

**Issue by gino0631** [JamesHeinrich/getID3#147]
Would it be possible to return a decoded video codec name (currently, only FourCC identifier is returned, while the mapping seems to be present in `QuicktimeVideoCodecLookup()`), as well as bit rates of audio and video streams, for MOV files like [this](https://github.com/gino0631/nextcloud-metadata/files/1705303/OO4A3570.MOV.zip)?

**Comment by JamesHeinrich** [comment]
Sure, it now returns $info['video']['fourcc_lookup'] if known.
https://github.com/JamesHeinrich/getID3/commit/598fdd5523076a2e9d6b5733ca4d3b6a3a93a605

**Comment by gino0631** [comment]
Thanks for fixing so quickly. And what about the bit rates of audio and video streams - do you have any thoughts how this can be implemented?

**Comment by JamesHeinrich** [comment]
I don't know how to properly read whatever metadata may be present that describes the bitrate of each audio/video stream. If someone does know how to implement it properly, let me know.

However, in the case of your example, PCM audio bitrate is trivial to calculate if we know the bits-per-sample, samples-per-second, and number-of-channels. This is now calculated for PCM audio streams.
And if we know the audio bitrate, and the overall bitrate (from the size of file and playtime) then we can assume that the video bitrate is the remainder of overall minus audio.

Changed in https://github.com/JamesHeinrich/getID3/commit/d3a72c99d10279345d29a97227677f5d8529f498

**Comment by gino0631** [comment]
I also don't know how to calculate the bit rate, but I found that [MediaInfo](https://mediaarea.net/en/MediaInfo) as well as FFmpeg do it somehow.

For the sample I've provided MediaInfo shows 34.6 Mb/s overall, 32.5 Mb/s for video, and 1536 kb/s for audio. FFmpeg shows 34596 kb/s overall, 32534 kb/s for video, and 1536 kb/s for audio.

Of course, source code of these tools has all the answers, but it is not easy to find that needle. According to my current understanding, the calculation is done by computing a sum of sample sizes from [`stsz`](https://wiki.multimedia.cx/index.php/QuickTime_container#stsz) (which is 6106408 in our case) and then dividing it by the duration of the stream (6106408 * 8 * 30000 / 45045 = 32534974). FFmpeg's implementation is [here](https://github.com/FFmpeg/FFmpeg/blob/master/libavformat/mov.c). Hope this helps.
